pkg/cmd/buildinformation/delete/delete.go

```diff
@@ -0,0 +1,124 @@
+package delete
+
+import (
+	"fmt"
+
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/OctopusDeploy/cli/pkg/cmd"
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/OctopusDeploy/cli/pkg/question"
+	"github.com/OctopusDeploy/cli/pkg/util/flag"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/buildinformation"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/client"
+	"github.com/spf13/cobra"
+)
+
+const (
+	FlagPackageId = "package-id"
+	FlagVersion   = "version"
+)
+
+type DeleteFlags struct {
+	PackageId *flag.Flag[string]
+	Version   *flag.Flag[string]
+}
+
+func NewDeleteFlags() *DeleteFlags {
+	return &DeleteFlags{
+		PackageId: flag.New[string](FlagPackageId, false),
+		Version:   flag.New[string](FlagVersion, false),
+	}
+}
+
+type DeleteOptions struct {
+	*cmd.Dependencies
+	*DeleteFlags
+	*question.ConfirmFlags
+	ID string
+}
+
+func NewDeleteOptions(id string, deleteFlags *DeleteFlags, confirmFlags *question.ConfirmFlags, dependencies *cmd.Dependencies) *DeleteOptions {
+	return &DeleteOptions{
+		Dependencies: dependencies,
+		DeleteFlags:  deleteFlags,
+		ConfirmFlags: confirmFlags,
+		ID:           id,
+	}
+}
+
+func NewCmdDelete(f factory.Factory) *cobra.Command {
+	deleteFlags := NewDeleteFlags()
+	confirmFlags := question.NewConfirmFlags()
+	cmd := &cobra.Command{
+		Use:     "delete <id>",
+		Short:   "Delete a build information",
+		Long:    "Delete a build information in Octopus Deploy",
+		Aliases: []string{"del", "rm", "remove"},
+		Example: heredoc.Docf(`
+			$ %[1]s build-information delete BuildInformation-1
+			$ %[1]s build-info rm BuildInformation-1
+			$ %[1]s build-info del --package-id ThePackage --version 1.2.3
+		`, constants.ExecutableName),
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				args = append(args, "")
+			}
+
+			opts := NewDeleteOptions(args[0], deleteFlags, confirmFlags, cmd.NewDependencies(f, c))
+
+			return deleteRun(opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&deleteFlags.PackageId.Value, deleteFlags.PackageId.Name, "p", "", "The Package ID of the build information to delete")
```

Comment:
Not a problem for this PR but the `p` short parameter is used for project name, pass phrase and package name/id depending on the command

Reply:
Yea, once you start allowing short hands there will be clashes :)